### PR TITLE
Add option to validate hosted domain from Google.

### PR DESCRIPTION
**Pull request by rkistner**
Counter-intuitively, the `hd` parameter passed by the provider is only a suggestion to simplify the client UX, and is never validated. We have to validate the domain when we receive the authentication response.

### CODE REVIEW

**Comment by drubin** [comment]
@rkistner nice fix! Can't we add a test for this param

File: README.md [hunk 1]
```diff
@@ -66,6 +66,10 @@ Sample config.ru:
         # The name must be `admin`.
         provider :google_oauth2, 'YOUR_KEY (client id)', 'YOUR_SECRET', name: 'admin',
             access_type: 'online', hd: 'yourdomain.com', approval_prompt: 'auto'
+
+        # IMPORTANT: To restrict logins to your domain, you have to configure the
```
Comment (drubin) [inline README.md]: You should also update the "Create an `config/initializers/admin_auth.rb` configuring your domain:" section to include the security part.

**Comment by rkistner** [comment]
@drubin Thanks, done.

**Comment by rkistner** [comment]
Released in v0.1.4

**Pull request merged**